lib/src/hpack/hpack.dart

```diff
@@ -49,7 +49,10 @@ class Header {
   Header(this.name, this.value, {this.neverIndexed = false});
 
   factory Header.ascii(String name, String value) {
-    return Header(ascii.encode(name), ascii.encode(value));
+    // Specs: `However, header field names MUST be converted to lowercase prior
+    // to their encoding in HTTP/2. A request or response containing uppercase
+    // header field names MUST be treated as malformed (Section 8.1.2.6).`
+    return Header(ascii.encode(name.toLowerCase()), ascii.encode(value));
```

Comment:
I think this was relying on callers to ensure it's lowercase - which most callers may readily guarantee as they construct the header with constant strings.

(i.e. it was a performance optimization to avoid calling `name.toLowerCase()` when it's already lower case).

Reply:
Hm, if performance is an issue we might add a parameter `[isLowerCase = false]` to the factory? But this might be a bit much..

Reply:
Probably fine to leave it like it is for now - one can always add an optional `isLowerCase` parameter later on.